execution_engine/src/system/mint/detail.rs

```diff
@@ -0,0 +1,39 @@
+use casper_types::{
+    system::{mint, mint::TOTAL_SUPPLY_KEY},
+    Key, U512,
+};
+
+use crate::system::mint::{runtime_provider::RuntimeProvider, storage_provider::StorageProvider};
+
+// Please do not expose this to the user!
+pub(crate) fn reduce_total_supply_unchecked<P>(
```

Comment:
nit: i think this is meant to be `reduce_total_supply_unsafe` ... as what is meant is the perms aren't enforced ... it actually does does do checked arithmetic ?

Reply:
you're right, corrected 👍